upstairs/src/lib.rs

```diff
@@ -1551,20 +1556,42 @@ impl Buffer {
         self.data.is_empty()
     }
 
-    pub fn write(&mut self, offset: usize, data: &[u8]) {
+    /// Writes data to the buffer, setting `owned` to `true`
+    ///
+    /// # Panics
+    /// - The offset must be block-aligned
+    /// - The data length must be divisible by block size
+    /// - Data cannot exceed the buffer's lengt
```

Comment:
I think my other comment (on the specific git rev before the rebase) made it out, but
I'm not sure.  Just in case, there is a typo here.